### PR TITLE
Fix Gazebo Skid Steer Model in Sim

### PR DESCRIPTION
**Pull request by ashviniyer21**
This change fixes issues with the gazebo controller, specifically the issue with linear and angular velocity appearing flipped. This also cleans up some of the sim config code.

### CODE REVIEW

**Comment by Isaac-F24** [comment]
This fixes the switch between linear and angular velocity for the tested systems. MPC runs successfully and with similar performance on my own computer and lunabotics laptop, with quite good performance. Only difference was due to mapping and avoiding obstacles, so that was temporarily disabled.

Changes:
Hardcoded new values to match the simulated bot (wheel seperation, wheel diameter)
Switched left wheel direciton (to account for flip) and right/left wheels (account for turning the right way)
Lowered friction (allows for easier turns w/ sliding friction

Temporarily turned MPC obstacle weight off due to inconsistent mapping to verify both systems performed the same.

**Comment by Isaac-F24** [comment]
The new changes provide a good balance between extremely slow turning and low friction (such that it doesn't look like we're on an ice rink). The turning is slightly slower than a _perfect_, environment but resembles the environment relatively well. Linear velocity is accurate, and MPC can follow paths well. [https://docs.google.com/presentation/d/1aBeNWQw2P2HW-tpmpk5T9W1i8QrbZwpnbIDSoVQLQxA/edit#slide=id.g2b94cf38523_0_45](here) is some testing data, and on the NAS are videos/csv files with the data as well.

File: lunabot_sim/dummy_bot_description/src/gazebo_ros_skid_steer_drive.cpp [hunk 1]
```diff
@@ -441,14 +445,19 @@ void GazeboRosSkidSteerDrive::FiniChild() {
...
-  wheel_speed_[LEFT_FRONT] = vr + va * wheel_separation_ / 2.0;
-  wheel_speed_[LEFT_REAR] = vr + va * wheel_separation_ / 2.0;
+  double angularWeight = 1.0;
+
+  wheel_speed_[LEFT_FRONT] = -(vr - angularWeight * va * wheel_separation_ / 2.0);
+  wheel_speed_[LEFT_REAR] = -(vr - angularWeight * va * wheel_separation_ / 2.0);
 
-  wheel_speed_[RIGHT_FRONT] = vr - va * wheel_separation_ / 2.0;
-  wheel_speed_[RIGHT_REAR] = vr - va * wheel_separation_ / 2.0;
+  wheel_speed_[RIGHT_FRONT] = vr + angularWeight * va * wheel_separation_ / 2.0;
```
[the remark this answers is not in the document]
Comment (Isaac-F24) [reply lunabot_sim/dummy_bot_description/src/gazebo_ros_skid_steer_drive.cpp]: yes

[the remark this answers is not in the document]
Comment (Isaac-F24) [reply lunabot_sim/dummy_bot_description/src/gazebo_ros_skid_steer_drive.cpp]: comment already was there but I wrote more

**Comment by Isaac-F24** [comment]
> The graphs from the slides honestly look kinda bad, I'm not sure if we're going to get much better though); I'll leave it up to your judgement to decide that. (if you any interesting videos that'd be cool but I think the graphs are more important). Added a couple of minor comments otherwise.

the simulated graphs are hard to get perfect- closer to the theoretical line is more slippery (looked like ice rink), and imo less like real life, so a little more friction was better

**Comment by raghavauppuluri13** [comment]
mpc_fix_gazebo_skid_steer
just for completeness. this is what mpc looks like. seems to look good for now!

**Review by raghavauppuluri13 — APPROVED** [review]

**Pull request merged**